### PR TITLE
Move reel_board default LVGL Kconfigs

### PR DESCRIPTION
**Pull request by vanwinkeljan**
As broughtup in #30464, move reel_board default LVGL Kconfigs from  `samples/display/lvgl/boards/reel_board.conf` to `boards/arm/reel_board/Kconfig.defconfig`

### CODE REVIEW

**Review by MaureenHelm — APPROVED** [review]

File: samples/display/lvgl/boards/reel_board.conf [hunk 1]
```diff
@@ -1,9 +0,0 @@
-CONFIG_SPI=y
```
Comment (gmarull) [inline samples/display/lvgl/boards/reel_board.conf:1]: How is SPI enabled now?

Comment (vanwinkeljan) [reply samples/display/lvgl/boards/reel_board.conf:1]: It was already enabled at board level:
https://github.com/zephyrproject-rtos/zephyr/blob/d5072d13786edc4bc34a169e743d5aeeee34a819/boards/arm/reel_board/reel_board_defconfig#L14

Same for the display:
https://github.com/zephyrproject-rtos/zephyr/blob/d5072d13786edc4bc34a169e743d5aeeee34a819/boards/arm/reel_board/Kconfig.defconfig#L29-L31

**Comment by vanwinkeljan** [comment]
rebased against latest master to take along fixes for ci failures

**Comment by gmarull** [comment]
To be honest, I think this strategy goes against porting rules (e.g. enabling SPI):

https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#general-recommendations

```
...
For consistency and to make it easier for users to build generic applications that are not board specific for your board, please follow these guidelines while porting.

Unless explicitly recommended otherwise by this section, leave peripherals and their drivers disabled by default.
...
```

So for example, if I take the "hello_world" sample I get:

```
Memory region         Used Size  Region Size  %age Used
           FLASH:       23224 B         1 MB      2.21%
            SRAM:        4488 B       256 KB      1.71%
        IDT_LIST:         120 B         2 KB      5.86%
```

If I compile the same sample with `CONFIG_SPI=n`, as it's not needed:

```
Memory region         Used Size  Region Size  %age Used
           FLASH:       19244 B         1 MB      1.84%
            SRAM:        4200 B       256 KB      1.60%
        IDT_LIST:          88 B         2 KB      4.30%
```

This does not affect LVGL settings location (sample or board), but in the end it is related because one may end up needing a board overlay (e.g. for SPI) if porting guide is followed.

**Comment by vanwinkeljan** [comment]
@gmarull I did not question why the `CONFIG_SPI` was there but there was clearly no need to add it once more.

Now looking at the porting guidelines it is cleat to my why it is there as it is to comply with following exceptions:
```
 * If your board uses a well-known connector standard (like Arduino, Mikrobus, Grove, or 96Boards connectors), add connector nodes to your DTS and configure pin muxes accordingly.
 * Configure components that enable the use of these pins, such as configuring an SPI instance to use the usual Arduino SPI pins.
```

It looks like the reel board supports an arduino connector so following he porting guidelines it needs to enable SPI by default.

https://github.com/zephyrproject-rtos/zephyr/blob/487af47f58e8dfe1af0257e9d41c8e119da07a72/boards/arm/reel_board/dts/reel_board.dtsi#L157-L163

**Comment by erwango** [comment]
> It looks like the reel board supports an arduino connector so following the porting guidelines it needs to enable SPI by default.

@vanwinkeljan. What guidelines state is all about configuration, so yes, board should provide default _configuration_. But only the minimum should be _enabled_. Pripherals and features enabling is application responsibility.
So SPI should be default n on reel_board.
If required for LVGL, it could be conditionally enabled.

**Review by erwango — CHANGES_REQUESTED** [review]
Please move SPI enabling under `if LVGL` condition

**Comment by vanwinkeljan** [comment]
@erwango  Note that CONFIG_SPI is not part of this PR and it was already set (else I would have done you suggestion).

As I have no idea what other areas are impacted I think it is better to raise a issue so that the boards maintainer can investigate why CONFIG_SPI is set.

**Review by erwango — APPROVED** [review]
@vanwinkeljan Creating a PR for SPI default seems reasonnable.

**Comment by gmarull** [comment]
Even though now SPI is now enabled, I'd add something like the following so that a removal of default SPI is easier later:

```
 config SPI 
 	default y 
 	depends on SSD16XX 
```

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**